src/node/setupServer.ts

```diff
@@ -1,15 +1,144 @@
 import { ClientRequestInterceptor } from '@mswjs/interceptors/lib/interceptors/ClientRequest'
 import { XMLHttpRequestInterceptor } from '@mswjs/interceptors/lib/interceptors/XMLHttpRequest'
-import { createSetupServer } from './createSetupServer'
-
+import { SetupApi } from '../createSetupApi'
+import { RequestHandler } from '../handlers/RequestHandler'
+import { LifeCycleEventsMap, SharedOptions } from '../sharedOptions'
+import { RequiredDeep } from '../typeUtils'
+import { mergeRight } from '../utils/internal/mergeRight'
+import { bold } from 'chalk'
+import { MockedRequest } from '../utils/request/MockedRequest'
+import { handleRequest } from '../utils/handleRequest'
+import {
+  IsomorphicResponse,
+  MockedResponse as MockedInterceptedResponse,
+} from '@mswjs/interceptors'
+import { devUtils } from '../utils/internal/devUtils'
 /**
  * Sets up a requests interception in Node.js with the given request handlers.
  * @param {RequestHandler[]} requestHandlers List of request handlers.
  * @see {@link https://mswjs.io/docs/api/setup-server `setupServer`}
  */
-export const setupServer = createSetupServer(
-  // List each interceptor separately instead of using the "node" preset
-  // so that MSW wouldn't bundle the unnecessary classes (i.e. "SocketPolyfill").
-  ClientRequestInterceptor,
-  XMLHttpRequestInterceptor,
-)
+// export const setupServer = createSetupServer(
+//   // List each interceptor separately instead of using the "node" preset
+//   // so that MSW wouldn't bundle the unnecessary classes (i.e. "SocketPolyfill").
+//   ClientRequestInterceptor,
+//   XMLHttpRequestInterceptor,
+// )
+
+export type ServerLifecycleEventsMap = LifeCycleEventsMap<IsomorphicResponse>
+
+const DEFAULT_LISTEN_OPTIONS: RequiredDeep<SharedOptions> = {
+  onUnhandledRequest: 'warn',
+}
+
+export class SetupServerApi extends SetupApi<ServerLifecycleEventsMap> {
+  private resolvedOptions: RequiredDeep<SharedOptions>
+
+  constructor(handlers: RequestHandler[]) {
+    super([ClientRequestInterceptor, XMLHttpRequestInterceptor], handlers)
+
+    this.resolvedOptions = {} as RequiredDeep<SharedOptions>
+
+    // TODO: Re-think this
+    this.init()
+  }
+
+  public async init(): Promise<void> {
+    const _self = this
+
+    this.interceptor.on('request', async function setupServerListener(request) {
```

Comment:
We don't need to keep this function's context, so this can be an arrow function ridding us of the `_self = this` assignment.

Reply:
I believe we've added this function simply to have a name to make these listeners easier to debug.

Reply:
Ok! Thank you.